src/profiles_management/helpers/kfam.py

```diff
@@ -0,0 +1,102 @@
+"""Utility module for manipulating KFAM resources."""
+
+import logging
+from typing import List
+
+from lightkube import Client
+from lightkube.generic_resource import GenericNamespacedResource, create_namespaced_resource
+from lightkube.resources.rbac_authorization_v1 import RoleBinding
+
+log = logging.getLogger(__name__)
+
+AuthorizationPolicy = create_namespaced_resource(
+    group="security.istio.io",
+    version="v1beta1",
+    kind="AuthorizationPolicy",
+    plural="authorizationpolicies",
+)
+
+
+def has_kfam_annotations(resource: GenericNamespacedResource) -> bool:
+    """Check if resource has "user" and "role" KFAM annotations."""
+    return (
+        resource.metadata
+        and resource.metadata.annotations
+        and "role" in resource.metadata.annotations
+        and "user" in resource.metadata.annotations
+    )
+
+
+def resource_is_for_profile_owner(resource: GenericNamespacedResource) -> bool:
+    """Check if the resource is for the Profile owner."""
+    if not resource.metadata:
```

Comment:
This is giving us slightly incorrect information. This will return False if there is no resource metadata, but that does not mean the profile is not of the owner.

Reply:
Could you elaborate? I didn't understand what you meant.

If a resource would have no metadata then it wouldn't be a resource of the owner in a Profile. Thus the function should return false

Reply:
So if `resource.metadata` is `None` then the RB or AP is automatically from the owner? I thought this was some sort of weird condition where either of those resources did not have metadata.

Reply:
Pushed a commit that unified how we handle the functions after our sync. Only `bool` is not returned
1377733

Small heads up that I had to slightly modify the signature, since I unpinned `pyright` and the LSP complained about the types. But should be good now.